src/Apps/Order/Routes/Shipping/useAddressAutocomplete.ts

```diff
@@ -0,0 +1,139 @@
+import { AutocompleteInputOptionType } from "@artsy/palette"
+import { Address } from "Components/AddressForm"
+import { useFeatureFlag } from "System/useFeatureFlag"
+import { useDebounce } from "Utils/Hooks/useDebounce"
+import { getENV } from "Utils/getENV"
+import { useCallback, useState } from "react"
+
+// NOTE: Due to the format of this key (a long string of numbers that cannot be parsed as json)
+// This key must be set in the env as a json string like SMARTY_EMBEDDED_KEY_JSON_JSON='{ "key": "xxxxxxxxxxxxxxxxxx" }'
+const smartyCreds = getENV("SMARTY_EMBEDDED_KEY_JSON") || { key: "" }
+console.log({ smartyCreds })
+const key = smartyCreds.key
+
+export type SuggestionJSON = {
+  city: string
+  entries: number
+  secondary: string
+  state: string
+  street_line: string
+  zipcode: string
+}
+
+export interface AddressSuggestion extends AutocompleteInputOptionType {
+  address: Omit<Address, "name">
+  entries: number
+}
+
+export const useAddressAutocomplete = () => {
+  const [result, setResult] = useState<SuggestionJSON[]>([])
+
+  // Add feature flag checks including country, etc here later
+  // console.log({ key, kind: typeof key })
+
+  const isAPIKeyPresent = !!key
+  const isFeatureFlagEnabled = !!useFeatureFlag("address_autocomplete_us")
+  const isFeatureEnabled = isAPIKeyPresent && isFeatureFlagEnabled
```

Comment:
When calling the hook, we might want to pass the search country in to select which api to use - and right now only US would be enabled.